returnn/util/file_cache.py

```diff
@@ -73,7 +74,6 @@ def __init__(
         self._cleanup_files_wanted_older_than_days = cleanup_files_wanted_older_than_days
         self._cleanup_disk_usage_wanted_free_ratio = cleanup_disk_usage_wanted_free_ratio
         self._touch_files_thread = _TouchFilesThread(cache_base_dir=self.cache_directory)
-        self._touch_files_thread.start()
```

Comment:
I would simply set `self._touch_files_thread: Optional[_TouchFilesThread] = None`, i.e. not create the thread object at all.

Reply:
Well ok, maybe not so relevant, maybe your current code is also fine.

Reply:
If we leave this none we need a check+init everywhere. I think the implementation as it is now is simpler in that regard. Also, shouldn't the thread object itself be cheap, and only allocate memory for the OS thread once started?

Reply:
> only allocate memory for the OS thread once started?

That was exactly the part I was not sure about. I guess it's also implementation specific to CPython. But probably it should be fine.